src/types.ts

```diff
@@ -47,15 +47,16 @@ export interface QueryOpts<TGraphQLVariables = OperationVariables> {
   context?: Record<string, any>;
 }
 
-export interface GraphqlQueryControls<TGraphQLVariables = OperationVariables> {
+export interface GraphqlQueryControls<TGraphQLVariables = OperationVariables, TData = any> {

```

Comment:
1. Alter the order to `<TData = any, TGraphQLVariables = OperationVariables>`
2. Rename `GraphqlQueryControls` to `QueryControls` as @danilobuerger mentioned.  While this is not intended to be exposed, it is public.  The best way to signal this parameterized ordering change to TS users is to change the name of the `interface`.  

It's a simple change and I am comfortable with the above strategy to be used without waiting for a major release cycle.